driver/modern_bpf/helpers/store/auxmap_store_params.h

```diff
@@ -1375,6 +1375,32 @@ static __always_inline void apply_dynamic_snaplen(struct pt_regs *regs, uint16_t
 		BPF_CORE_READ_INTO(&port_remote, sk, __sk_common.skc_dport);
 		port_local = ntohs(port_local);
 		port_remote = ntohs(port_remote);
+
+		if(port_remote == 0 && sockaddr != NULL)
+		{
+			BPF_CORE_READ_USER_INTO(&socket_family, (struct sockaddr*)sockaddr, sa_family);
+
+			// If socket_family is 0 we skip this part.
+			switch(socket_family)
+			{
+				case AF_INET:
+				{
+					struct sockaddr_in sockaddr_in = {};
+					bpf_probe_read_user(&sockaddr_in, bpf_core_type_size(struct sockaddr_in), sockaddr);
+					port_remote = ntohs(sockaddr_in.sin_port);
+					break;
+				}
+				case AF_INET6:
+				{
+					struct sockaddr_in6 sockaddr_in6 = {};
+					bpf_probe_read_user(&sockaddr_in6, bpf_core_type_size(struct sockaddr_in6), sockaddr);
+					port_remote = ntohs(sockaddr_in6.sin6_port);
+					break;
+				}
+				default:
+					break;
+			}
+		}
 	}
```

Comment:
```suggestion
		if(port_remote == 0 && sockaddr != NULL)
		{
			if(socket_family == AF_INET)
			{
				struct sockaddr_in sockaddr_in = {};
				bpf_probe_read_user(&sockaddr_in, bpf_core_type_size(struct sockaddr_in), sockaddr);
				port_remote = ntohs(sockaddr_in.sin_port);
			}
			else
			{
				struct sockaddr_in6 sockaddr_in6 = {};
				bpf_probe_read_user(&sockaddr_in6, bpf_core_type_size(struct sockaddr_in6), sockaddr);
				port_remote = ntohs(sockaddr_in6.sin6_port);
			}
		}
	}
```

Reply:
something like this should be enough, WDYT?

Reply:
Actually I think can be worth add the `AF_UNIX` case, wdyt?

Reply:
uhm we cannot fall in the `AF_UNIX` case here... if you take a look at the code above we are already inside  `if(socket_family == AF_INET || socket_family == AF_INET6)`